packages/core/src/routes/hook.test.ts

```diff
@@ -0,0 +1,250 @@
+import {
+  HookEvent,
+  type Hook,
+  type HookEvents,
+  type HookConfig,
+  type CreateHook,
+} from '@logto/schemas';
+import { pickDefault } from '@logto/shared/esm';
+
+import {
+  mockCreatedAtForHook,
+  mockHook,
+  mockHookList,
+  mockNanoIdForHook,
+  mockTenantIdForHook,
+} from '#src/__mocks__/hook.js';
+import { MockTenant } from '#src/test-utils/tenant.js';
+import { createRequester } from '#src/utils/test-utils.js';
+
+const { jest } = import.meta;
+
+const hooks = {
+  findAllHooks: async (): Promise<Hook[]> => mockHookList,
+  insertHook: async (data: CreateHook): Promise<Hook> => ({
+    ...mockHook,
+    ...data,
+  }),
+  findHookById: async (): Promise<Hook> => mockHook,
+  updateHookById: async (_: unknown, data: Partial<CreateHook>): Promise<Hook> => ({
+    ...mockHook,
+    ...data,
+  }),
+  deleteHookById: jest.fn(),
+};
+
+const tenantContext = new MockTenant(undefined, { hooks });
+
+const hookRoutes = await pickDefault(import('./hook.js'));
+
+describe('hook routes', () => {
+  const hookRequest = createRequester({ authedRoutes: hookRoutes, tenantContext });
+
+  it('GET /hooks', async () => {
+    const response = await hookRequest.get('/hooks');
+    expect(response.status).toEqual(200);
+    expect(response.body).toEqual(mockHookList);
+    expect(response.header).not.toHaveProperty('total-number');
```

Comment:
What is the total-number used for here?

Reply:
This is to ensure we don't use pagination for this route.